### ISSUE TITLE
Possible bug in latest dev Theano install with nvidia driver 378.13

### ISSUE THREAD

**Issue by mpgussert** [fchollet/keras#5750]
Greetings all!

I don't know if this is coming from Theano or Keras, but I will be posting this on their Issues page as well.

Forcing driver 378.13 for Theano causes softmax not to normalize on the GPU.  Here's the code

```
import numpy as np
import keras
from keras.models import Sequential, Model
from keras.layers import Input, Dense, Convolution2D, MaxPooling2D, UpSampling2D, Cropping2D, Flatten
from keras.regularizers import l2, l1, l1l2
from keras import backend as K

def atariPolicyModel(state, actions):
  model = Sequential()
  model.add(Convolution2D(16,8,8,subsample=(4,4), input_shape=state,  border_mode='same', activation='relu'))
  model.add(Convolution2D(32,4,4,subsample=(2,2), border_mode='same', activation='relu'))
  model.add(Flatten())
  model.add(Dense(256, activation='relu'))
  model.add(Dense(actions, activation='softmax'))
  return model
  
state = (4,210,160)
actions = 8

In = np.random.rand(1,4,210,160)

model1 = atariPolicyModel(state, actions)

print np.sum(model1.predict(In))

```

In Ubuntu 16.04...

```
mpgussert@Cthulhu:KerasAC$ THEANO_FLAGS=device=cuda0 ipython -i Tests.py 
Python 2.7.12 (default, Nov 19 2016, 06:48:10) 
Type "copyright", "credits" or "license" for more information.

IPython 5.1.0 -- An enhanced Interactive Python.
?         -> Introduction and overview of IPython's features.
%quickref -> Quick reference.
help      -> Python's own help system.
object?   -> Details about 'object', use 'object??' for extra details.
Using Theano backend.
WARNING: loading blacklisted driver because the load was forced.
Using cuDNN version 5105 on context None
Mapped name None to device cuda0: Quadro GP100 (0000:02:00.0)
1.63243

In [1]: exit
mpgussert@Cthulhu:KerasAC$ ipython -i Tests.py 
Python 2.7.12 (default, Nov 19 2016, 06:48:10) 
Type "copyright", "credits" or "license" for more information.

IPython 5.1.0 -- An enhanced Interactive Python.
?         -> Introduction and overview of IPython's features.
%quickref -> Quick reference.
help      -> Python's own help system.
object?   -> Details about 'object', use 'object??' for extra details.
Using Theano backend.
1.0

In [1]: exit

```

can anyone else verify this behavior?

Thanks!

**Comment by patyork** [comment]
`WARNING: loading blacklisted driver because the load was forced.` should be a hint.

**Comment by mpgussert** [comment]
I'm aware, but I'm not sure that's the issue.  I've forced drivers with theano before and I haven't seen it fail in this manner (usually it's a very loud error or warning).  Unfortunately I need to use this driver with my card :( 

It also should be noted that this doesn't happen in TF, so I'm fairly certain it's either an issue in theano or an issue with how Keras handles the theano backend.

**Comment by patyork** [comment]
A major reason certain drivers were blacklisted was not just that certain ops didn't exist (and would error) but also that certain ops would give incorrect results (e.g. 2+2=5).

In addition, the message `WARNING: loading blacklisted driver because the load was forced.` itself was removed recently (last week or so, as better compatibility was added) leading me to think that Theano/libgpuarray is not up to date either.

**Comment by mpgussert** [comment]
Ok! I will update my crap and try again.  As this is a theano issue I will continue discussion of this in my other post on their issue page.

Thank you!
